packages/react/src/Label/Label.tsx

```diff
@@ -8,15 +8,36 @@ import clsx from 'clsx'
 import { forwardRef } from 'react'
 import type { ForwardedRef, LabelHTMLAttributes, PropsWithChildren } from 'react'
 
+type LabelHintProp = {
+  /** Provides additional context to the user, for example whether the associated form field is optional or required. Will disregard the value of the optional prop if provided with a text. */
+  hint?: string
+}
+
+type LabelOptionalProp = {
+  /** Whether the associated input is optional. Will append the text '(niet verplicht)' to the label if no hint is provided. */
+  optional?: boolean
+}
+
+export type LabelProps = PropsWithChildren<LabelHTMLAttributes<HTMLLabelElement>> & (LabelHintProp & LabelOptionalProp)
+
+const LabelHint = ({ hint }: LabelHintProp) => <span className="ams-hint">({hint})</span>
```

Comment:
Making this a separate `Hint` component matches the CSS component to a React one and prevents repetition between the Label and FieldSet components.

Reply:
Indeed! Creating a new Hint component.